src/Functions/keyvaluepair/src/KeyValuePairExtractorBuilder.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <unordered_set>
+#include <Functions/keyvaluepair/src/impl/LazyEscapingKeyValuePairExtractor.h>
+#include <Functions/keyvaluepair/src/impl/SimpleKeyValuePairEscapingProcessor.h>
+#include <Functions/keyvaluepair/src/impl/state/KeyStateHandler.h>
+#include <Functions/keyvaluepair/src/impl/state/ValueStateHandler.h>
+#include "KeyValuePairExtractor.h"
+
+namespace DB
+{
+
+template <typename Response = std::unordered_map<std::string, std::string>>
+class KeyValuePairExtractorBuilder
```

Comment:
Consider making a simple struct with public fields and setting them explicitly in caller without `with...` methods (if the amount of code won't much greater)

Reply:
I honestly feel like a Builder presents a much nicer & fluent interface. Plus, it's a very well known & established pattern. Do you think we could keep this as the way it is or it's a no go for you?

Reply:
I honestly feel like a Builder presents a much nicer & fluent interface. Plus, it's a very well known & established pattern. Do you think we could keep this as the way it is or it's a no go for you?

Reply:
Ok, let's keep it